src/utils/upload.js

```diff
@@ -70,8 +76,8 @@ export default class Upload {
 
     this.progress({ indeterminate: true });
 
-    if (this.upload.behavior && uploadBehaviors[this.upload.behavior]) {
-      return new uploadBehaviors[this.upload.behavior](this).resume();
```

Comment:
This is the bug I referred to in the PR description. The way I saw it manifest was that any time an S3 multipart upload was made to fail at any chunk beyond the first, or during the finalize step, the resume did not pick up properly and tried to start at the beginning. [We now remember the behavior state](https://github.com/girder/girder_web_components/pull/7/files#diff-8af87c0ce07db8082e10e648f84bda61R66) on the upload itself so that the state isn't lost each time `resume()` is called.

Reply:
This made me think: we could get away with an even simpler API on the Upload class of just `start()` rather than `start()` and `resume()`. It's less expressive and powerful, but simpler and less constrained in how it should be used. If you guys prefer that, I'm happy to change it.